staging_service/import_specifications/individual_parsers.py

```diff
@@ -0,0 +1,165 @@
+"""
+Contains parser functions for use with the file parser framework.
+"""
+
+import pandas
+import re
+
+# TODO update to C impl when fixed: https://github.com/Marco-Sulla/python-frozendict/issues/26
+from frozendict.core import frozendict
+from pathlib import Path
+from typing import TextIO
+
+from staging_service.import_specifications.file_parser import (
+    PRIMITIVE_TYPE,
+    ErrorType,
+    SpecificationSource,
+    Error,
+    ParseResult,
+    ParseResults,
+)
+
+# this version defines the schema of the xsv and Excel files.
+# if we need to support a new format, use the version number to detect whether to use
+# the current parsing strategy or the new strategy that you will develop.
+_VERSION = 1
+
+_DATA_TYPE = "Data type:"
+_VERSION_STR = "Version:"
+_HEADER_SEP = ";"
+_EXPECTED_HEADER = f"{_DATA_TYPE} <data_type>{_HEADER_SEP} {_VERSION_STR} <version>"
+_HEADER_REGEX = re.compile(f"{_DATA_TYPE} (\\w+){_HEADER_SEP} {_VERSION_STR} (\\d+)")
+
+
+class _ParseException(Exception):
+    pass
+
+
+def _parse_header(line: str, spec_source: SpecificationSource, maximum_version: int) -> str:
+    # return is data type
+    match = _HEADER_REGEX.fullmatch(line)
+    if not match:
+        raise _ParseException(Error(
+            ErrorType.PARSE_FAIL,
+            f'Invalid header; got "{line}", expected "{_EXPECTED_HEADER}"',
+            spec_source
+        ))
+    version = int(match[2])
+    if version > maximum_version:
+        raise _ParseException(Error(
+            ErrorType.PARSE_FAIL,
+            f"Schema version {version} is larger than maximum processable "
+            + f"version {maximum_version}",
+            spec_source
+        ))
+    return match[1]
+
+
+def _get_datatype(input_: TextIO, spec_source: SpecificationSource, maximum_version: int) -> str:
+    # return is data type
```

Comment:
What does this comment mean?

Reply:
It's cruft from when the return type was more complicated. Removed